script.js

```diff
@@ -0,0 +1,177 @@
+const searchBtn = document.querySelector('.search-button')
+const clearBtn = document.querySelector('.clear-button')
+
+searchBtn.addEventListener('click', (event) =>{
+    event.preventDefault()
+    errorDisplay()
+    resetDisplay()
+    getWeatherInfo()
+    .then(displayWeatherInfo)
+    .catch(onError)
+})
+
+clearBtn.addEventListener('click', (event) =>{
+    event.preventDefault()
+    clearSearch()
+})
+
+form = document.querySelector('form')
+const daysInput = document.querySelector('.num-days')
+form.addEventListener('change',() =>{
+    const searchInput = document.getElementById('weather-type').value
+    if(searchInput === 'forecast'){
+        daysInput.style.display = 'block'
+    }
+    else{
+        daysInput.style.display = 'none'
+    }
+})
+
+async function getWeatherInfo() {
+    try {
+        const searchInput = document.getElementById('location').value
+        const weatherChoice = document.getElementById('weather-type').value
+        const days = document.getElementById('num-days').value
+        const weatherResults = document.querySelector('.weather-result')
+        const url = 'https://api.weatherapi.com/v1/'
+        const myKey = config.API_KEY
+    
+        if (!searchInput) {
+            weatherResults.style.display = 'none'
+            throw new Error('No search input provided.')
+        }
+    
+        if (weatherChoice) {
+            const response =  await fetch( url + `${weatherChoice}.json?` + `key=${myKey}` + `&q=${searchInput}` + `&days=${days}`)
+            const updateddata = getFetchedData(response)
+            return updateddata
+        } 
+    } catch (error) {
+        console.log(error)
+        throw error
+    }
+}
+
+function getFetchedData(response) {
+    if (!response.ok) {
+        const errorData = response.json()
+        throw new Error(errorData.error.message)
+    }
+    return response.json()
+}
+
+function displayWeatherInfo(weatherInfo) {
```

Comment:
This function is too long. Please consider splitting it. Can show the current weather and the forecast in different functions.